### PR TITLE
[Grouping&Mapping] Group Properties Fixes

### PR DESCRIPTION
**Pull request by Guillar1**
- Fixed aspect properties not being generated correctly.
- Added support for the missing path and name properties from external source.
- Unit tests

### CODE REVIEW

File: packages/itwin/grouping-mapping-widget/src/widget/components/GroupPropertyUtils.ts [hunk 1]
```diff
@@ -158,6 +158,10 @@ const extractNested = (propertyTraversal: string[], propertyFields: Field[]): Pr
           nestedContentField.pathToPrimaryClass &&
           nestedContentField.pathToPrimaryClass.length > 1
         ) {
+          // Hardcoded navigation to external source repository metadata.
+          if (nestedContentField.contentClassInfo.name === "BisCore:RepositoryLink") {
```
Comment (MindaugasButkus) [inline packages/itwin/grouping-mapping-widget/src/widget/components/GroupPropertyUtils.ts:162]: Could you add unit tests for `GroupPropertyUtils.ts`?

Comment (Guillar1) [reply packages/itwin/grouping-mapping-widget/src/widget/components/GroupPropertyUtils.ts:162]: In another PR.

**Review by bingjiez — APPROVED** [review]

File: packages/itwin/grouping-mapping-widget/src/test/GroupPropertyUtils.test.ts [hunk 2]
```diff
@@ -0,0 +1,680 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import type { ECProperty } from "@itwin/insights-client";
+import { DataType } from "@itwin/insights-client";
+import type { PropertyMetaData } from "../widget/components/GroupPropertyUtils";
+import { findProperties } from "../widget/components/GroupPropertyUtils";
+import { convertToECProperties } from "../widget/components/GroupPropertyUtils";
+import { convertPresentationFields } from "../widget/components/GroupPropertyUtils";
```
Comment (MindaugasButkus) [inline packages/itwin/grouping-mapping-widget/src/test/GroupPropertyUtils.test.ts:10]: Importing from `GroupPropertyUtils` multiple times.

**Review by MindaugasButkus — APPROVED** [review]

**Comment by Guillar1** [comment]
@aruniverse can you review?

File: common/config/rush/pnpm-lock.yaml [hunk 3]
```diff
@@ -339,26 +339,26 @@ importers:
...
-      '@itwin/core-quantity': 3.2.2_@itwin+core-bentley@3.2.2
-      '@itwin/core-react': 3.2.2_635ad49d75618b72dca44ab860e4509b
-      '@itwin/eslint-plugin': 3.2.2_eslint@7.32.0+typescript@4.3.5
-      '@itwin/imodel-components-react': 3.2.2_fd0453885139ba43494b554614c626b0
-      '@itwin/presentation-common': 3.2.2_aa7c17727124c31a56baeab89140d952
-      '@itwin/presentation-components': 3.2.2_0c8e54e1d46986e00a4f386734b921be
-      '@itwin/presentation-frontend': 3.2.2_7d67946d87cba22f9e077d73ef83df22
-      '@itwin/webgl-compatibility': 3.2.2
+      '@itwin/appui-abstract': 3.4.4_@itwin+core-bentley@3.4.4
+      '@itwin/appui-layout-react': 3.4.4_6c2c54728238f2c24c3c3de97dc854de
```
Comment (aruniverse) [inline common/config/rush/pnpm-lock.yaml:343]: Why was the lockfile updated?

**Review by aruniverse — COMMENTED** [review]
Why was the lockfile updated?
If you meant to pull in a newer version of itwinjs for your PKG and are using those apis please update the pkg.json

**Pull request merged**